library/Fission/Web/Auth/Types.hs

```diff
@@ -1,17 +1,36 @@
 -- | Authorization types; primarily more semantic aliases
 module Fission.Web.Auth.Types
-  ( ExistingUser
-  , HerokuAddOnAPI
+  ( HerokuAddOnAPI
+  , RegisterDid
+  , HigherOrder
+  , Token (..)
+  , BasicToken (..)
+  , BearerToken (..)
   ) where
 
 import Database.Esqueleto
 import Servant (BasicAuth)
+import Servant.API.Experimental.Auth
+import Servant.Server.Experimental.Auth
 
 import Fission.Prelude
 import Fission.Models
 
--- | Authorization check for a normal user
-type ExistingUser = BasicAuth "existing user" (Entity User)
-
 -- | Authorization check for the Heroku Addon API
 type HerokuAddOnAPI = BasicAuth "heroku add-on api" ByteString
+
+-- | Authorization check to return encoded did for registering new users
+type RegisterDid = AuthProtect "register-did"
+type instance AuthServerData (AuthProtect "register-did") = ByteString
```

Comment:
🤔 Have you considered a `newtype` for the DID auth? It'll likely save us from orphans now or down the road.

Reply:
I think I would just do `DID` because those will be used throughout our system and not always in the context of auth
```haskell
newtype DID  = DID { unwrap :: ByteString }

type instance AuthServerData (AuthProtect "register-did") = DID
```